hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/MessagesPublishingStartupValidationHook.java

```diff
@@ -0,0 +1,121 @@
+        this.topicsCache = topicsCache;
+        this.producer = brokerMessagesProducer;
+    }
+
+    @Override
+    public void accept(ServiceLocator serviceLocator) {
+        logger.info("Validating publishing messages to broker is available");
+        Duration delay = Duration.of(config.getLongProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_RETRY_INTERVAL), ChronoUnit.MILLIS);
+        Duration timeout = Duration.of(config.getLongProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_TIMEOUT_MS), ChronoUnit.MILLIS);
+        RetryPolicy<BrokerMessagesProducingResults> retryPolicy = retryPolicy(delay, timeout);
+        Fallback<BrokerMessagesProducingResults> fallback = fallback();
+        Failsafe.with(fallback, retryPolicy)
+                .with(scheduler)
+                .get(context -> {
+                    logger.info("Trying to publish messages, attempt:{}", context.getAttemptCount());
+                    return publishMessagesToBroker();
+                });
+    }
+
+    private BrokerMessagesProducingResults publishMessagesToBroker() {
+        return topicsCache
+                .getTopic(config.getStringProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_TOPIC_NAME))
+                .map(it -> producer.publishMessages(
+                        it,
+                        validationMessages(config.getIntProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_MESSAGES_COUNT)),
+                        config.getLongProperty(Configs.BROKER_PUBLISHING_STARTUP_VALIDATION_TIMEOUT_MS)
+                ))
+                .orElseThrow(() -> new IllegalStateException("Missing topic to validate publishing messages on startup"));
+    }
+
+    private Fallback<BrokerMessagesProducingResults> fallback() {
+        return Fallback.of((CheckedConsumer<ExecutionAttemptedEvent<? extends BrokerMessagesProducingResults>>) event -> {
+            throw new PublishingStartupValidationException(event.getLastResult());
+        })
+                .handle(BrokerMessagesProducingException.class)
+                .handleIf((validationResults, throwable) -> validationResults.isFailure());
+    }
+
+    private RetryPolicy<BrokerMessagesProducingResults> retryPolicy(Duration delay, Duration maxDuration) {
+        return new RetryPolicy<BrokerMessagesProducingResults>()
+                .handle(BrokerMessagesProducingException.class)
+                .withMaxRetries(-1)
+                .withDelay(delay)
+                .withMaxDuration(maxDuration)
+                .handleIf((result, t) -> result.isFailure());
+    }
+
+    private List<Message> validationMessages(int number) {
+        return IntStream.range(0, number).mapToObj(it -> {
+                    String messageId = MessageIdGenerator.generate();
+                    long timestamp = clock.millis();
+                    byte[] messageContent = messageContentWrapper.wrapJson("validation-message".getBytes(defaultCharset()), messageId, timestamp, Collections.emptyMap());
```

Comment:
I think we could set `"validation-message".getBytes(defaultCharset())` in a static variable to not repeat this operation over and over :)